### PR TITLE
Rails 7.1/7.2

### PR DESCRIPTION
**Pull request by jprosevear**

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -2,20 +2,16 @@ name: Unit Tests
 on:
   pull_request:
     branches:
-      - '*'
+      - "*"
   push:
     branches:
-      - '*'
+      - "master"
```
Comment (jprosevear) [inline .github/workflows/unit_tests.yml:8]: Avoid building when pushing to PR branches

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -2,20 +2,16 @@ name: Unit Tests
...
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         include:
-          - gemfile: gemfiles/rails_3.1.x.gemfile
-            ruby-version: 2.2
-          - gemfile: gemfiles/rails_3.2.x.gemfile
-            ruby-version: 2.2
```
Comment (jprosevear) [inline .github/workflows/unit_tests.yml:18]: The CI setup just seems to generally fail with these two versions now (ie ruby setup can't handle it)

File: arel-helpers.gemspec [hunk 3]
```diff
@@ -16,7 +16,7 @@ Gem::Specification.new do |s|
 
   s.add_development_dependency 'appraisal'
   s.add_development_dependency 'combustion', '~> 1.3'
-  s.add_development_dependency 'database_cleaner', '~> 1.8'
+  s.add_development_dependency 'database_cleaner', '~> 2.0'
```
Comment (jprosevear) [inline arel-helpers.gemspec:19]: Needed for 7.1/7.2 - pinned the 1.8 in the older gemfiles

File: lib/arel-helpers/join_association.rb [hunk 4]
```diff
@@ -262,6 +264,33 @@ def join_association_6_1_0(table, association, join_type, options = {})
         end
       end
 
+      def join_association_7_2_0(table, association, join_type, options = {})
+        aliases = options.fetch(:aliases, []).index_by(&:table_name)
+        associations = association.is_a?(Array) ? association : [association]
+
+        alias_tracker = ActiveRecord::Associations::AliasTracker.create(
+          table.connection_pool, table.name, {}
```
Comment (jprosevear) [inline lib/arel-helpers/join_association.rb:272]: This is the only update line - AliasTracker.create takes the connection pool now:

https://github.com/rails/rails/pull/51725/files#diff-6b3375555f1d9c06188a2676ec95c585b1e3c464c3d0bb22a0170d794575cb55

and 

https://github.com/rails/rails/pull/51725/files#diff-18a561656864ea240daf46bdb1f50faace49f9ef74b90bcf667d9bbb17fce084R954

File: .github/workflows/unit_tests.yml [hunk 5]
```diff
@@ -38,8 +34,12 @@ jobs:
             ruby-version: 3.0
           - gemfile: gemfiles/rails_7.0.x.gemfile
             ruby-version: 3.0
+          - gemfile: gemfiles/rails_7.1.x.gemfile
+            ruby-version: 3.1
+          - gemfile: gemfiles/rails_7.2.x.gemfile
+            ruby-version: 3.3
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
```
Comment (jprosevear) [inline .github/workflows/unit_tests.yml:42]: Avoid GH Actions deprecation warnings

File: .github/workflows/unit_tests.yml (hunk 2 above)
Comment (camertron) [reply .github/workflows/unit_tests.yml:18]: Ok fine to remove then 👍

File: .github/workflows/unit_tests.yml [hunk 6]
```diff
@@ -38,8 +34,12 @@ jobs:
             ruby-version: 3.0
           - gemfile: gemfiles/rails_7.0.x.gemfile
             ruby-version: 3.0
+          - gemfile: gemfiles/rails_7.1.x.gemfile
```
Comment (camertron) [inline .github/workflows/unit_tests.yml:37]: Can we add an entry for 3.2 as well?

**Review by camertron — APPROVED** [review]
Awesome, thank you! 🎉

File: .github/workflows/unit_tests.yml (hunk 6 above)
Comment (jprosevear) [reply .github/workflows/unit_tests.yml:37]: Done.

**Comment by jprosevear** [comment]
> Awesome, thank you! 🎉

I believe workflow needs to be re-approved to pass before it can be merged.

**Comment by jprosevear** [comment]
@camertron could you re-approve the workflow

**Comment by camertron** [comment]
Published in v2.15.0

**Pull request merged**